src/main/java/org/deepsymmetry/Wayang.java

```diff
@@ -265,4 +276,51 @@ public static void sendFrame() {
             }
         }
     }
+
+    /**
+     * Send a frame of pixels async, corresponding to whatever has been drawn in the image returned by open(),
+     * to the display.
+     *
+     * @throws LibUsbException       if there is a problem communicating.
+     * @throws IllegalStateException if the Push 2 has not been opened.
+     */
+    public static void sendFrameAsync() {
+        if (transferBuffer == null) {
+            throw new IllegalStateException("Push 2 device has not been opened");
+        }
+        IntBuffer transferred = IntBuffer.allocate(1);
+        transfer = LibUsb.allocTransfer();
+        LibUsb.fillBulkTransfer(transfer, pushHandle, (byte) 0x01, headerBuffer, LibUsb::freeTransfer,  null, 1000);
```

Comment:
Here we would supply our own callback that would call `LibUsb.freeTransfer()` and then the callback that was passed to `sendFrameAsync()`. Or if you don’t think there is any need to inform callers of the results of the frame transfer attempt, we could just use a logging framework, as discussed below.